src/Equinox.EventStoreDb/EventStoreDb.fs

```diff
@@ -168,6 +168,12 @@ module private Write =
 
 module private Read =
     open FSharp.Control
+    let private processReadStreamResult (res: EventStoreClient.ReadStreamResult) f onStreamNotFound = task {
+        match! res.ReadState with
+        | ReadState.StreamNotFound -> return onStreamNotFound
+        | ReadState.Ok -> return! f res
```

Comment:
note for .NET Enum values, matching like this produces a warning to indicate that there is a potential for an out of range enum value. While I generally would not do a catchall, I think that makes the most sense for a case like this, and it removes the warning.
(You could also layer an Active Pattern over the enum to codify that - will try that now)

Reply:
Dang, these evil .NET enums.. 
When layering an active pattern over the enum wouldn't you encounter the same issue just at a different position? (At the declaration of active pattern). But it might still be better and more reusable.

Reply:
Main issue with that active pattern is that it allocates, and would be crazy to do in a tight loop.
As you say, you still have a catchall that would need to be revisited if they ever added a (meaningful) new enum value.
Given there are only 2 usages, it would definitely be overkill here - the only value would be that you get to encapsulate and DRY the mapping of relevant cases.

Reply:
Ah ok I see! Didn't know about the performance downsides of active patterns (i'm still pretty new to the f# world). Thanks again for the insights!